modules/serverless-sinkbinding-yaml.adoc

```diff
@@ -0,0 +1,150 @@
+// Module included in the following assemblies:
+//
+// serverless/knstive_eventing/serverless-sinkbinding.adoc
+
+[id="serverless-sinkbinding-yaml_{context}"]
+= Using SinkBinding with the YAML method
+
+ This guide describes the steps required to create, manage and delete a SInkBinding instance using YAML files.
+
+.Prerequisites
+
+* You will need to have a Knative Serving and Eventing installation.
+* You will need to have the `kn` CLI installed.
+
+.Procedure
+
+. To check that SinkBinding is set up correctly, create a Knative event display service, or event sink, that dumps incoming messages to its log.
++
+Copy the following sample YAML into a file named `service.yaml`:
++
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  template:
+    spec:
+      containers:
+        - image: gcr.io/knative-releases/github.com/knative/eventing-contrib/cmd/event_display
+----
++
+After you have created the `service.yaml` file, apply it by entering the following command:
++
+----
+$ oc apply --filename service.yaml
+----
+
+. Create a SinkBinding that will direct events to the service.
++
+You can do this by creating a file named `sinkbinding.yaml` and copying the following sample code into it:
++
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: SinkBinding
+metadata:
+  name: bind-heartbeat
+spec:
+  subject:
+    apiVersion: batch/v1
+    kind: Job <1>
+    selector:
+      matchLabels:
+        app: heartbeat-cron
+
+  sink:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+  ceOverrides:
+    extensions:
+      sink: bound
+----
++
+<1> In this example, any Job with the label `app: heartbeat-cron` will be bound to the event sink.
++
+After you have created the `sinkbinding.yaml` file, apply it by entering the following command:
++
+----
+$ oc apply --filename sinkbinding.yaml
+----
+
+. Create a CronJob.
++
+You can do this by creating a file named `heartbeats-source.yaml` and copying the following sample code into it:
++
+----
```

Comment:
Ah, here is the missing piece which is needed for the `kn` command.

@matzew what do you think, wouldn't it be better to use a `Deployment` instead of a `CronJob` as an example here ? Mostly because the `CronJob` is tricky because the sink binding needs to point to the included Job (not the CronJob itself), which makes it kind of hard to understant. That is not the case for a regular Deployment (and tbh I think Deployments are a much more likely use case then a CronJob).

The question is whether we can craft a `Deployment` descriptor with e.g. the heartbeat image.

Reply:
@rhuss I'm not sure I completely agree with that analysis.  With both a `deployment` as well as the `CronJob` the sinkbinding duck-types the underlying `container` partial schema.  Hence the 'podspecable' name.  I think this example works.  If anything perhaps we just make that more explicit that the sinkbinding will end up targeting the resulting pod?

Reply:
Actually, one thing that could further emphasize/clarify this point is printing out the sinkbinding itself, ie:

```
oc get sinkbinds bind-heartbeat -oyaml
...
spec:
  ceOverrides:
    extensions:
      sink: bound
  sink:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: event-display
      namespace: default
  subject:
    apiVersion: batch/v1
    kind: Job
    namespace: default
    selector:
      matchLabels:
        app: heartbeat-cron
```
which shows the sink, as well as the subject the sinkbinding is acting on

Reply:
Should I remove `ceOverrides` from this too @lberk if it's not being included in the YAML or `kn` commands now?

Reply:
Ya if you're not going to use the `ceOverrides` it'd make sense to remove it here too